versioned/persist/inmem/src/main/java/org/projectnessie/versioned/persist/inmem/InmemoryDatabaseAdapter.java

```diff
@@ -252,4 +257,69 @@ protected RefLog fetchFromRefLog(NonTransactionalOperationContext ctx, Hash refL
         .map(ProtoSerialization::protoToRefLog)
         .collect(Collectors.toList());
   }
+
+  @Override
+  protected void writeAttachments(Stream<Entry<AttachmentKey, AttachmentValue>> attachments) {
+    attachments.forEach(
+        e -> store.attachments.put(dbKey(e.getKey().toByteString()), e.getValue().toByteString()));
+  }
+
+  @Override
+  protected boolean consistentWriteAttachment(
+      AttachmentKey key, AttachmentValue value, Optional<String> expectedVersion) {
+    try {
+      store.attachments.compute(
+          dbKey(key.toByteString()),
+          (k, current) -> {
+            if (expectedVersion.isPresent()) {
+              if (current != null) {
+                try {
+                  AttachmentValue currentValue = AttachmentValue.parseFrom(current);
+                  if (currentValue.hasVersion()
+                      && currentValue.getVersion().equals(expectedVersion.get())) {
+                    return value.toByteString();
+                  }
+                } catch (InvalidProtocolBufferException e) {
+                  throw new RuntimeException(e);
+                }
+              }
+              // consistent-update failed, key not present or hash!=expected, throw
+              throw new IllegalStateException();
```

Comment:
please add some message to this exception. I think it can be even:
`throw new IllegalStateException(consistent-update failed, key not present or hash!=expected)`

Reply:
It's a bit of a hack here. Adding a different approach to make it a bit clearer.